### DISCUSSION TITLE
Massive feedback noises

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#54] [Q&A]
@marcel-licence Since I was unable to reload samples from SD card I though I could give a try to your new updated code. I have downloaded and uploaded your new code to my ESP32 Audio-kit, but it also doesn't solved my problem. But when ESP32 Audio-kit boots up massive feedback noises are heard and even the noise of me clicking on the buttons can be heard with loud volume. But when I re-upload the old version of your code (without as5600 and es8388 support)
There is not such feedback noises.
By the way the original problem did not get solved either.

**Comment by marcel-licence** [comment]
`#define SAMPLER_ALWAYS_PASS_THROUGH /* can be used to pass line in through audio processing to output */`
is set in the config.h

**Reply by souravlal** [reply]
This is disabled in my code.

**Reply by souravlal** [reply]
@marcel-licence can you send a picture of the settings of your Arduino in tools menu. Because I want to know whether I made a mistake there, because also I am still unable to reload sample from sd card.

**Reply by souravlal** [reply]
Do any one facing a similar problem?

**Reply by marcel-licence** [reply]
This has been already solved in: https://github.com/marcel-licence/esp32_midi_sampler/discussions/48

**Reply by souravlal** [reply]
No, I am still getting the feedback noises with your new code.

**Reply by marcel-licence** [reply]
Please try to add:
`ES8388_SetIn2OoutVOL(0, 0);`
Into your setup() routine. That disables the pass through of the ES8388. There are also more possibilities. I highly recommend reading the datasheet.

**Reply by souravlal** [reply]
Actually I am using the AC101 version with your new code not the es8388 version.

**Reply by marcel-licence** [reply]
Is the feedback generated when the mainloop is running?
You can put #if 0 ... #endif around the code. There are two options -> the AC101 routes the input signal to the output or the code handles the input and forward it to the output. You can also ensure that input instead of mic input is selected during startup.

**Reply by souravlal** [reply]
@marcel-licence When the ESP32 Audio-kit starts up the feedback noises are also starts to appear, it continues to do so until I power it off.
Where should I put  ' #if 0 ... #endif ' code?
Where should I check whether mic input is selected or not?

**Comment by marcel-licence** [comment]
`    ac101_setSourceMic();` <- source is mic, you can also activate line in ac101_setSourceLine
You can set `master_output_gain = 0.0f;` that should mute the internal processing.
Please look into the code esp32_midi_sampler.ino
`       /* apply master_output_gain */
        fl_sample[n] *= master_output_gain;
        fr_sample[n] *= master_output_gain;`

Following is reading the samples from input:
`    i2s_read_stereo_samples_buff(fl_sample, fr_sample, SAMPLE_BUFFER_SIZE);`

And following is writing samples to output:
`    if (i2s_write_stereo_samples_buff(fl_sample, fr_sample, SAMPLE_BUFFER_SIZE))`
